docs/guides/blocks/sync-tables/dynamic.md

```diff
@@ -1,7 +1,428 @@
+
+Creating a dynamic sync table can be a bit of an involved process, and it requires an in-depth understanding of the target API and [sync table schemas][schemas_sync_table]. The sections below cover the components of a dynamic sync table definition.
+
+
+### Define the table
+
+The dynamic sync table is defined using the [`addDynamicSyncTable()`][addDynamicSyncTable] method. The `name` is user visible and should follow the [naming conventions][sync_tables_naming] of regular sync tables. The `description` is not required but recommended.
+
+```ts
+pack.addDynamicSyncTable({
+  name: "Tasks",
+  description: "The tasks in the selected project.",
+  listDynamicUrls: async function (context) {
+    // Return the datasets the user has access to.
+    // ...
+  },
+  getName: async function (context) {
+    // Return the name of the selected dataset.
+    // ...
+  },
+  getDisplayUrl: async function (context) {
+    // Return a display URL for the selected dataset.
+    // ...
+  },
+  getSchema: async function (context) {
+    // Return a schema for the selected dataset.
+    // ...
+  },
+  formula: {
+    // Sync the rows from the dataset.
+    // ...
+  },
+});
+```
+
+More information the the various components of the dynamic sync table are described in the sections below.
+
+!!! bug
+    It is a known issue that dynamic sync tables assume and require user authentication. Ensure that the Pack has a form of user authentication defined and that the `connectionRequirement` of the sync table is configured to use it (the default).
+    <!-- https://golinks.io/bug/19815 -->
+    <!-- https://golinks.io/bug/19713 -->
+    <!-- https://golinks.io/bug/19710 -->
+
+
+### Generate the URL list
+
+The [`listDynamicUrls`][listDynamicUrls] function is responsible for generating the list of datasets that the user can select from. Each dataset is represented by a unique URL, known as the dynamic URL. This list is typically generated by querying an external API using a connected account.
+
+The function should return an array of [`MetadataFormulaObjectResultType`][MetadataFormulaObjectResultType] objects. The `display` field should be set to the user-facing name of the dataset, and the `value` field to its URL.
+
+```ts
+pack.addDynamicSyncTable({
+  // ...
+  listDynamicUrls: async function (context) {
+    let response = await context.fetcher.fetch({
+      method: "GET",
+      url: "https://api.example.com/projects",
+    });
+    let projects = response.body.items;
+    let results = [];
+    for (let project of projects) {
+      results.push({
+        display: project.name,
+        value: project.links.self,
+      });
+    }
+    return results;
+  },
+  // ...
+});
+```
+
+The URL you choose to represent the dataset is arbitrary, and is not directly exposed to the user. Most Packs choose to use an API endpoint URL for ease of development. The URL shown to users is generated using the [`getDisplayUrl`](#display-url) method.
```

Comment:
I think there might be better framing than saying the URL is arbitrary. It's true that you can put arbitrary strings for the dynamic url and that might be worth noting. But the spirit of the field is that you use it to specify an API entity to sync from, and the value very directly instructs your sync implementation on where to get data. So in that sense it's not arbitrary, in the common case the value needs to be something quite specific, though its form can vary as desired.

Reply:
Fair enough, I've removed the use of the word "arbitrary" and rewritten it a little.